src/metrics.rs

```diff
@@ -38,10 +39,10 @@ pub trait LocalMetric {
 pub trait MayFlush: LocalMetric {
     /// If the LocalMetric is already flushed in last `flush_interval_sec` seconds, then do nothing,
     /// else flush and update last flush time.
-    fn try_flush(&self, last_flush: &Cell<Instant>, flush_interval: coarsetime::Duration) {
-        let now = Instant::recent();
+    fn try_flush(&self, last_flush: &Cell<u64>, flush_interval: Duration) {
+        let now = timer::recent_millis();
         let last_tick = last_flush.get();
-        if now.duration_since(last_tick) < flush_interval {
+        if Duration::from_millis(now - last_tick) < flush_interval {
```

Comment:
better not to construct `Duration` since it uses slow multiplications.. `flush_interval` can be `flush_interval_ms`.